.github/workflows/ci.yml

```diff
@@ -7,7 +7,7 @@ on:
 env:
   node_version: "14.x"
   tf_version: "0.14.7" # must match value in terraform-iac/*/app/main.tf
-
+concurrency: ${{ github.base_ref }}
```

Comment:
I thought we were thinking of putting this on just the Terraform Plan job.

Reply:
I was originally planning on that, but this note scared me

> Note: When concurrency is specified at the job level, order is not guaranteed for jobs or runs that queue within 5 minutes of each other.

Do we want to ensure that the last commit to a PR actually gets planned?

Reply:
I assumed that was to account for the way jobs don't always start right away (and so therefore we have a race condition). I feel like I'd prefer most of our CI to run rather than having it all blocked on a deployment.

Reply:
Worst case scenario, "previous" jobs that were already blocked get cancelled and we re-run CI. That doesn't sound too bad.

Actually, I guess the real worst-case scenario is if we have a deployment blocked on CI and we update our PR that had CI running, we cancel the deployment. We'd then have to re-run the deployment. Not a problem, but still annoying.

Reply:
Hopefully, the Slack notification would make it obvious to a developer that their queued deployment was cancelled.

Reply:
I think worst case is that somebody pushes several commits to a PR in succession, the last commit contains a mistake that breaks their infrastructure, but the CI plan doesn't catch the problem because the wrong run survives. So the broken infrastructure gets deployed.

I don't know how likely this is to happen. But 5 minutes is a long time when you're pushing small changes fast.

Reply:
> When a concurrent job or workflow is queued, if another job or workflow using the same concurrency group in the repository is in progress, the queued job or workflow will be pending. Any previously pending job or workflow in the concurrency group will be canceled.

([source](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#jobsjob_idconcurrency))

> the CI plan doesn't catch the problem

I'm not sure I understand the concern here. If CI is cancelled, that shows up as "failing" CI. If CI is pending, it won't show up as green until it's finished running.

Reply:
Check out the `update prd files` commit [here](https://github.com/byu-oit/casfsy/commits/main) on April 20th. You can see it marked as ❌ because the deployment was cancelled (without being re-run).

Reply:
If CI is cancelled, I'd expect developers to re-run it on their PR to get a ✔ before merging the PR.